spec/acceptance/command_shell_spec.rb

```diff
@@ -0,0 +1,533 @@
+            wait_for_expect do
+              expect(payload.size).to be > 0
+            end
+
+            console.sendline payload.handler_command(default_module_datastore: default_module_datastore)
+            console.recvuntil(/Started reverse TCP handler[^\n]*\n/)
+            payload_process = executed_payload
+            session_id = nil
+
+            # Wait for the session to open, or break early if the payload is detected as dead
+            larger_retry_count_for_powershell = 600
+            wait_for_expect(larger_retry_count_for_powershell) do
+              unless payload_process.alive?
+                break
+              end
+
+              session_opened_matcher = /session (\d+) opened[^\n]*\n/
+              session_message = ''
+              begin
+                session_message = console.recvuntil(session_opened_matcher, timeout: 1)
+              rescue Acceptance::ChildProcessRecvError
+                # noop
+              end
+
+              session_id = session_message[session_opened_matcher, 1]
+              expect(session_id).to_not be_nil
+            end
+
+            [payload_process, session_id]
+          end
+
+          # @param [String] path The file path to read the content of
+          # @return [String] The file contents if found
+          def get_file_attachment_contents(path)
+            return 'none resent' unless File.exist?(path)
+
+            content = File.binread(path)
+            content.blank? ? 'file created - but empty' : content
+          end
+
+          before :each do |example|
+            next unless example.respond_to?(:parameter)
+
+            # Add the test environment metadata to the rspec example instance - so it appears in the final allure report UI
+            test_environment.each do |key, value|
+              example.parameter(key, value)
+            end
+          end
+
+          after :all do
+            driver.close_payloads
+            console.reset
+          end
+
+          context "#{Acceptance::Session.current_platform}" do
+            describe "#{Acceptance::Session.current_platform}/#{command_shell_runtime_name} command shell successfully opens a session for the #{payload_config[:name].inspect} payload" do
+              it(
+                "exposes available metasploit commands",
+                if: (
+                  # Assume that regardless of payload, staged/unstaged/etc, the command shell will have the same commands available
+                  # So only run this test when config_index == 0
+                  payload_config_index == 0 && Acceptance::Session.supported_platform?(payload_config)
+                  # Run if ENV['SESSION'] = 'java php' etc
+                  Acceptance::Session.run_session?(command_shell_config) &&
+                    # Only run payloads / tests, if the host machine can run them
+                    Acceptance::Session.supported_platform?(payload_config)
+                )
+              ) do
```

Comment:
This `it` block/test can be deleted entirely, command shells won't expose meterpreter compatibility information